src/cacheClient.ts

```diff
@@ -0,0 +1,35 @@
+import NodeCache from '@cacheable/node-cache';
+import { GenerateCacheKeyFn, HyphenEvaluationContext, HyphenProviderOptions } from './types';
+import { cache } from './config';
+
+export class CacheClient {
+  private readonly cache: NodeCache;
+  private readonly generateCacheKeyFn: GenerateCacheKeyFn;
+
+  constructor(cacheConfig?: HyphenProviderOptions['cache']) {
+    const ttlSeconds = cacheConfig?.ttlSeconds ?? cache.ttlSeconds;
+
+    this.cache = new NodeCache({
+      stdTTL: ttlSeconds,
+      checkperiod: ttlSeconds * 2,
+    });
+
+    this.generateCacheKeyFn = cacheConfig?.generateCacheKey ?? this.defaultGenerateCacheKey;
+  }
+
+  private defaultGenerateCacheKey(context: HyphenEvaluationContext): string {
+    return JSON.stringify(context);
```

Comment:
what happens if context is a very large object? will things overflow will it cause the cache to ballon in size? 

Is this the best way to generate a key? should we use something like https://www.npmjs.com/package/object-hash instead? ideally the way a key is calculated across all our provider languages is the same...

Reply:
Sorry, wrong push.